src/main/java/com/firebolt/jdbc/exception/ServerError.java

```diff
@@ -0,0 +1,295 @@
+        }
+
+        public String getQueryId() {
+            return queryId;
+        }
+
+        public String getRequestId() {
+            return requestId;
+        }
+
+        public String getQueryLabel() {
+            return queryLabel;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            Query query = (Query) o;
+            return Objects.equals(queryId, query.queryId) && Objects.equals(requestId, query.requestId) && Objects.equals(queryLabel, query.queryLabel);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(queryId, requestId, queryLabel);
+        }
+
+        @Override
+        public String toString() {
+            return "Query{" +
+                    "queryId='" + queryId + '\'' +
+                    ", requestId='" + requestId + '\'' +
+                    ", queryLabel='" + queryLabel + '\'' +
+                    '}';
+        }
+    }
+
+    public static class Error {
+        private final String code;
+        private final String name;
+        private final Severity severity;
+        private final Source source;
+        private final String description;
+        private final String resolution;
+        private final String helpLink;
+        private final Location location;
+
+        public Error(String code, String name, Severity severity, Source source, String description, String resolution, String helpLink, Location location) {
+            this.code = code;
+            this.name = name;
+            this.severity = severity;
+            this.source = source;
+            this.description = description;
+            this.resolution = resolution;
+            this.helpLink = helpLink;
+            this.location = location;
+        }
+
+        Error(JSONObject json) {
+            this(json.optString("code", null), json.optString("name", null),
+                    json.optEnum(Severity.class, "severity"),
+                    ofNullable(json.optString("source", null)).map(Source::fromText).orElse(Source.Unknown),
+                    json.optString("description", null), json.optString("resolution", null), json.optString("helpLink", null),
+                    ofNullable(json.optJSONObject("location", null)).map(Location::new).orElse(null));
+        }
+
+        public String getCode() {
+            return code;
+        }
+
+        public String getName() {
+            return name;
+        }
+
+        public Severity getSeverity() {
+            return severity;
+        }
+
+        public Source getSource() {
+            return source;
+        }
+
+        public String getDescription() {
+            return description;
+        }
+
+        public String getResolution() {
+            return resolution;
+        }
+
+        public String getHelpLink() {
+            return helpLink;
+        }
```

Comment:
Can we use something like lombok's [getters](https://projectlombok.org/features/GetterSetter) to avoid writing out every single parameter here?
Also possibly `@EqualsAndHashCode`

Reply:
We can but I  planned to remove Lombok, so it looks strange to continue using it.

Reply:
You are right, it does not affect the package wight but IMHO sometimes it makes code worse. I can explain more but if you really want I can add the annotation.

Reply:
Right, so let's use it appropriately in places where it doesn't make the code worse. Here for example it will be beneficial IMO.
Let's do big design decisions (like removing some dependencies) properly, with a doc or ticket describing the reasoning that's also shared and discussed with the team.

Reply:
After a long discussion I have added Lombok annotations there